airbyte-integrations/connectors/source-recurly/source_recurly/streams.py

```diff
@@ -0,0 +1,219 @@
+        :func:`read_records`. For more details:
+            https://developers.recurly.com/api/v2021-02-25/#section/Getting-Started/Pagination#query-parameters
+
+        :return: The cursor field(s) to be used in the `incremental` sync mode.
+        :rtype: Union[str, List[str]]
+        """
+        return self.DEFAULT_CURSOR
+
+    def read_records(
+        self,
+        sync_mode: SyncMode,
+        cursor_field: List[str] = None,
+        stream_slice: Mapping[str, any] = None,
+        stream_state: Mapping[str, Any] = None,
+    ) -> Iterable[Mapping[str, Any]]:
+        """
+        The method to be called to retrieve records from the Recurly API. It uses the Recurly Python client.
+        Resources having different logic (such as the `export_dates`) can override this method
+
+        :return: Iterable of dictionaries representing the Recurly resource
+        :rtype: Iterable
+        """
+        params = {"order": "asc", "sort": self.sort_key}
+
+        self.begin_time = (stream_state and stream_state[self.cursor_field]) or self.begin_time
+
+        if self.begin_time:
+            params.update({self.BEGIN_TIME_PARAM: self.begin_time})
+
+        # Call the Recurly client methods
+        items = getattr(self._client, self.client_method_name)(params=params).items()
+
+        for item in items:
+            yield self._item_to_dict(item)
+
+    def get_updated_state(self, current_stream_state: MutableMapping[str, Any], latest_record: Mapping[str, Any]):
+        """
+        Compares the current stream state cursor with the latest record cursor value and returns the latest or the most
+        recent cursor value (either the current cursor value or the latest record cursor value depending which of those
+        is the maximum).
+
+        :return: The value of the new current value of the cursor
+        :rtype: dict
+        """
+        current_updated_at = (current_stream_state or {}).get(self.cursor_field, "")
+        latest_record_updated_at = latest_record[self.cursor_field].isoformat()
+
+        return {self.cursor_field: max(latest_record_updated_at, current_updated_at)}
+
+    def _item_to_dict(self, resource):
+        """
+        Recursively converts the Recurly resource object to `dict`
+        """
+        if isinstance(resource, dict):
+            return dict((key, self._item_to_dict(value)) for key, value in resource.items())
+        elif hasattr(resource, "__iter__") and not isinstance(resource, str):
+            return [self._item_to_dict(value) for value in resource]
+        elif hasattr(resource, "__dict__"):
+            return dict([(key, self._item_to_dict(value)) for key, value in resource.__dict__.items()])
+        else:
+            return resource
+
+
+class RecurlyAccountsStream(BaseRecurlyStream):
+    name = "accounts"
+
+
+class RecurlyAccountCouponRedemptionsStream(BaseRecurlyStream):
```

Comment:
Rename to `AccountCouponRedemptions` this way you don't need to specify a name and is more close to the pattern followed by other connectors. Change all stream classes name you made.

Reply:
Done in https://github.com/airbytehq/airbyte/pull/8675/commits/4c08ce71753804df5ac0803759ffc8ea5d24af7d